<?xml version="1.0"?>
<launch>

    <arg name="experimentId" default="1" />

    <group ns="uav1">

        <!-- MAVROS -->
        <include file="$(find mavros)/launch/px4.launch">
            <arg name="fcu_url" value="/dev/serial0:57600"/>
            <arg name="gcs_url" value=""/>
            <arg name="tgt_system" value="1"/>
            <arg name="tgt_component" value="1"/>
        </include>

        <!-- worker node -->
        <node pkg="multi_uav_se_mission" name="worker_node" type="worker_node" output="screen">
            <param name="uavId" value="1" />
            <param name="altitude" value="4.0" />
            <param name="taskTypes" value="0,1" />
            <param name="taskAssignAtSeconds" value="10" />
            <param name="serialPort" value="/dev/ttyUSB0" />
            <param name="baud" value="57600" />
            <remap from="statistics" to="/uav1/statistics/data" />
            <remap from="statisticsLog" to="/uav1/statistics/log" />
            <remap from="statisticsMessages" to="/uav1/statistics/messages" />
            <remap from="statisticsTasks" to="/uav1/statistics/tasks" />
        </node>

        <!-- rosbag -->
        <node pkg="rosbag" type="record" name="record" args="-o $(find multi_uav_se_mission)/bag/uav1_$(arg experimentId) -a -x (.*)/camera(.*)"/>

    </group>

</launch>
